server/src/main/java/io/crate/metadata/Functions.java

```diff
@@ -69,9 +74,39 @@ public Functions copyOf() {
         return functions;
     }
 
-    @Inject
-    public Functions(Map<FunctionName, List<FunctionProvider>> functionImplementationsBySignature) {
-        this.functionImplementations = functionImplementationsBySignature;
+    public static class Builder {
+
+        private HashMap<FunctionName, List<FunctionProvider>> providersByName = new HashMap<>();
+
+        public void put(Signature signature, FunctionFactory factory) {
```

Comment:
Afaik this won't replace a factory for the same signature? If so, wouldn't `put` be a bit misleading?

```suggestion
        public void add(Signature signature, FunctionFactory factory) {
```

Reply:
Good point. Renamed it.